src/components/Profile/MyPosts/MyPosts.js

```diff
@@ -9,12 +11,12 @@ const MyPosts = (props) => {
   const newPostElement = React.createRef();
 
   const addPost = () => {
-    props.addPost();
+    props.dispatch(addPostActionCreator());
```

Comment:
Consider reading about mapDIspatchToProps function in Redux.

Reply:
So far this is not an Redux. Soon